credentials/jwt.go

```diff
@@ -0,0 +1,54 @@
+package credentials
+
+import (
+	"github.com/dgrijalva/jwt-go"
+	"io"
+	"net/url"
+	"strings"
+	"time"
+)
+
+type jwtProvider struct {
+	creds JwtCredentials
+}
+
+type claims struct {
+	jwt.StandardClaims
+}
+
+func (provider *jwtProvider) Retrieve() (io.Reader, error) {
+	expirationTime := provider.GetAppropriateExpirationTime()
+	tokenString, err := provider.BuildClaimsToken(expirationTime, provider.creds.URL, provider.creds.ClientId, provider.creds.ClientUsername)
+	if err != nil {
+		return nil, err
+	}
+
+	form := url.Values{}
+	form.Add("grant_type", string(jwtGrantType))
+	form.Add("assertion", tokenString)
+
+	return strings.NewReader(form.Encode()), nil
+}
+
+func (provider *jwtProvider) URL() string {
+	return provider.creds.URL
+}
+
+func (provider *jwtProvider) GetAppropriateExpirationTime() int64 {
+	return time.Now().Add(3 * time.Minute).Unix()
```

Comment:
is there any data behind the 3 minute time?  Might want to make the timeout a constant for naming.

Reply:
@g8rswimmer Hey! The 3 minute was a suggestion from some developer from SalesForce in their forum. Though, I cannot find the thread right now. I'll move it to a constant either way.